src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/fly/modes/FlyGeneric.kt

```diff
@@ -69,6 +71,71 @@ internal object FlyVanilla : Choice("Vanilla") {
 
 }
 
+internal object FlyCreative : Choice("Creative") {
+
+    override val parent: ChoiceConfigurable<*>
+        get() = ModuleFly.modes
+
+    private val speed by float("Speed", 0.1f, 0.1f..5f)
+
+    private object SprintSpeed : ToggleableConfigurable(this, "SprintSpeed", true) {
+        val speed by float("Speed", 0.1f, 0.1f..5f)
+    }
+
+    init {
+        tree(SprintSpeed)
+    }
+
+    private val maxVelocity by float("MaxVelocity", 4f, 1f..20f)
+
+    private val bypassVanillaCheck by boolean("BypassVanillaCheck", true)
+
+    private val forceFlight by boolean("ForceFlight", true)
+
+    override fun enable() {
+        player.abilities.allowFlying = true;
+    }
+
+    private fun shouldFlyDown(): Boolean {
+        if (!bypassVanillaCheck) return false
+        if (player.age % 40 != 0) return false
+
+        // check if the player is above a block or in mid-air
+        // if the player is right above a block, we don't need to fly down
+        if (world.getStatesInBox(player.boundingBox.offset(0.0, -0.55, 0.0)).anyMatch { !it.isAir }) return false
+
+        return true
+    }
+
+    val repeatable = repeatable {
+        player.abilities.flySpeed =
+            if (mc.options.sprintKey.isPressed && SprintSpeed.enabled) SprintSpeed.speed else speed
+
+        if (forceFlight) player.abilities.flying = true
+
+        if (player.velocity.lengthSquared() > maxVelocity * maxVelocity * maxVelocity) {
```

Comment:
This line says:
$v_\text {curr}^2 > v_\text {max}^3$

Are you sure that this is as planned?

Reply:
Ah yes, that's correct. I will correct it as soon as possible.